test/Confluent.Kafka.IntegrationTests/Tests/Producer_CustomPartitioner.cs

```diff
@@ -0,0 +1,140 @@
+                            break;
+                        case 1:
+                            // Default custom partitioner
+                            producerBuilder = new ProducerBuilder<string, string>(producerConfig);
+                            producerBuilder.SetDefaultPartitioner((string topicName, int partitionCount, ReadOnlySpan<byte> keyData, bool keyIsNull) =>
+                            {
+                                Assert.Equal(topic.Name, topicName);
+                                var keyString = System.Text.UTF8Encoding.UTF8.GetString(keyData.ToArray());
+                                return int.Parse(keyString.Split(" ").Last()) % partitionCount;
+                            });
+                            break;
+                        case 2:
+                            // Default custom partitioner in case where default topic config is present due to topic level config in top-level config.
+                            var producerConfig2 = new ProducerConfig
+                            {
+                                BootstrapServers = bootstrapServers
+                            };
+                            producerConfig2.Set("queuing.strategy", "lifo");
+                            producerBuilder = new ProducerBuilder<string, string>(producerConfig2);
+                            producerBuilder.SetDefaultPartitioner((string topicName, int partitionCount, ReadOnlySpan<byte> keyData, bool keyIsNull) =>
+                            {
+                                Assert.Equal(topic.Name, topicName);
+                                var keyString = System.Text.UTF8Encoding.UTF8.GetString(keyData.ToArray());
+                                return int.Parse(keyString.Split(" ").Last()) % partitionCount;
+                            });
+                            break;
+                        default:
+                            Assert.True(false);
+                            break;
+                    }
+
+                    using (var producer = producerBuilder.Build())
+                    {
+                        for (int i=0; i<PARTITION_COUNT; ++i)
+                        {
+                            producer.Produce(
+                                topic.Name,
+                                new Message<string, string> { Key = $"test key {i}", Value = $"test val {i}" }, dh);
+                        }
+
+                        producer.Flush(TimeSpan.FromSeconds(10));
+                    }
+                }
+            }
+
+
+            // Null key
+
+            using (var topic = new TemporaryTopic(bootstrapServers, PARTITION_COUNT))
+            using (var producer = new ProducerBuilder<Null, string>(producerConfig)
+                .SetDefaultPartitioner((string topicName, int partitionCount, ReadOnlySpan<byte> keyData, bool keyIsNull) =>
+                {
+                    Assert.True(keyIsNull);
+                    Assert.True(!keyIsNull);
```

Comment:
How can both of these be true?

Reply:
no idea how the second assert got there. i was testing the test at one point is all i can think of.